### PR TITLE
Chore/require authentication and scoping everywhere

### PR DESCRIPTION
**Pull request by zspencer**

### CODE REVIEW

File: app/policies/furniture_placement_policy.rb [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+  class Scope
+    attr_accessor :person, :scope
+    def initialize(person, scope)
+      @person = person
+      @scope = scope
+    end
+
+    def resolve
+      # TODO: is this even the right scope?
+      @scope.joins(:room).where(room: { space: @person.spaces })
```
Comment (zspencer) [inline app/policies/furniture_placement_policy.rb:14]: We probably want to use the `RoomPolicy::Scope` to do this: `where(room: RoomPolicy::Scope.new(person, Room.all).resolve)`

File: app/policies/room_policy.rb [hunk 2]
```diff
@@ -17,4 +17,17 @@ def create?
...
+  class Scope
+    attr_accessor :actor, :scope
+
+    def initialize(actor, scope)
+      self.actor = actor
+      self.scope = scope
+    end
+
+    def resolve
+      scope.accessable_by(actor)
```
[the remark this answers is not in the document]
Comment (anaulin) [reply app/policies/room_policy.rb]: Done.

File: app/policies/furniture_placement_policy.rb (hunk 1 above)
Comment (anaulin) [reply app/policies/furniture_placement_policy.rb]: Done.

File: app/policies/invitation_policy.rb [hunk 3]
```diff
@@ -16,4 +16,18 @@ def create?
   alias update? create?
   alias destroy? create?
   alias new? create?
+
+  class Scope
```
[the remark this answers is not in the document]
Comment (anaulin) [reply app/policies/invitation_policy.rb]: Done.

**Review by anaulin — APPROVED** [review]
Zee told me to approve this.

**Pull request merged**